web/stats_controller.go

```diff
@@ -0,0 +1,124 @@
+package web
+
+import (
+	"fmt"
+	"github.com/asdine/storm"
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/services"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/smartcontractkit/chainlink/store/presenters"
+	"go.uber.org/multierr"
+)
+
+// StatsController fetches high-level information about the nodes usage
+type StatsController struct {
+	App *services.ChainlinkApplication
+}
+
+// Show displays Stats, one page at a time
+// Example:
+//  "<application>/stats?size=1&page=2"
+func (sc *StatsController) Show(c *gin.Context) {
+	store := sc.App.Store
+	jobs := []models.JobSpec{}
+
+	size, page, offset, err := ParsePaginatedRequest(c.Query("size"), c.Query("page"))
+	if err != nil {
+		c.AbortWithError(422, err)
+		return
+	}
+
+	skip := storm.Skip(offset)
+	limit := storm.Limit(size)
+
+	if count, err := sc.App.Store.Count(&models.JobSpec{}); err != nil {
+		c.AbortWithError(500, fmt.Errorf("error getting count of JobSpec: %+v", err))
+	} else if err := sc.App.Store.AllByIndex("CreatedAt", &jobs, skip, limit); err != nil {
+		c.AbortWithError(500, fmt.Errorf("error getting Jobs: %+v", err))
+	} else if jss, err := sc.allJobStats(jobs); err != nil {
+		c.AbortWithError(500, fmt.Errorf("error getting job stats: %+v", err))
+	} else if account, err := store.KeyStore.GetAccount(); err != nil {
+		publicError(c, 400, err)
+	} else {
+		a := presenters.AccountBalance{
+			Address: account.Address.Hex(),
+		}
+		s := presenters.Stats{
+			Account:      a,
+			JobSpecStats: jss,
+		}
+
+		buffer, err := NewPaginatedResponse(*c.Request.URL, size, page, count, s)
+		if err != nil {
+			c.AbortWithError(500, fmt.Errorf("failed to marshal document: %+v", err))
+		} else {
+			c.Data(200, MediaType, buffer)
+		}
+	}
+}
+
+func (sc *StatsController) allJobStats(jobs []models.JobSpec) ([]presenters.JobSpecStats, error) {
```

Comment:
This logic doesn't seem like it has to be related to the controller. Are you up for making this its own model? Benefits would be that it could be reused other places, and you could test the logic more directly instead of talking to it through the controller.